cfme/common/provider.py

```diff
@@ -244,9 +249,9 @@ def update(self, updates, endpoints=None, cancel=False, validate_credentials=Tru
 
             # filling endpoints
             if endpoints:
-                endpoints = endpoints if isinstance(endpoints, list) else [endpoints, ]
+                endpoints = self._prepare_endpoints(endpoints)
 
-                for endpoint in endpoints:
+                for _, endpoint in endpoints.items():
```

Comment:
good point. because when I was writing for statement, I didn't exactly know whether I'd use key :)